src/Middlewares/OperationDefinitionNodeStore.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+namespace Enjin\Platform\Middlewares;
+
+use Closure;
+use GraphQL\Executor\ExecutionResult;
+use GraphQL\Language\AST\OperationDefinitionNode;
+use GraphQL\Language\Parser;
+use GraphQL\Type\Schema;
+use Rebing\GraphQL\Support\ExecutionMiddleware\AbstractExecutionMiddleware;
+use Rebing\GraphQL\Support\OperationParams;
+
+class OperationDefinitionNodeStore extends AbstractExecutionMiddleware
+{
+    /**
+     * The current request's Operation Definition Node.
+     */
+    public static OperationDefinitionNode $operationDefinitionNode;
+
+    /**
+     * Handle's middleware logic.
+     */
+    public function handle(string $schemaName, Schema $schema, OperationParams $params, $rootValue, $contextValue, Closure $next): ExecutionResult
+    {
+        $documentNode = Parser::parse($params->query);
+        $operationName = $params->operation;
+        static::$operationDefinitionNode = collect($documentNode->definitions)
+            ->when($operationName, fn ($collection) => $collection->where('name.value', '=', $operationName))
+            ->first();
+
+        return $next($schemaName, $schema, $params, $rootValue, $contextValue);
+    }
+
+    /**
+     * Get the current request's operation name.
+     */
+    public static function getOperationName()
+    {
+        return static::$operationDefinitionNode
```

Comment:
Should we check for null here incase operationDefinitionNode doesn't have value

Reply:
Not sure it would be possible to have a null value here would it?  I.e. there should always be at least one query or mutation that gets sent in every request.